### PR TITLE
Add cluster setup documentation

### PR DESCRIPTION
**Pull request by mrocklin**
This moves and reorganizes the content previously in

-  http://distributed.readthedocs.io/en/latest/setup.html
-  http://dask.pydata.org/en/latest/scheduler-choice.html

### CODE REVIEW

**Comment by mrocklin** [comment]
cc @electronwill if you have time and interest.  This could use review at a high level.

File: docs/source/setup/cli.rst [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+from manually SSH-ing into all of the machines to more automated systems like
+SGE/SLURM/Torque or Yarn/Mesos. Additionally, cluster SSH tools exist to send
+the same commands to many machines.  We recommend searching online for "cluster
+ssh" or "cssh"..
+
+
+API
+---
+
+These may be out-dated.  We recommend referring to the ``--help`` text of your
```
Comment (TomAugspurger) [inline docs/source/setup/cli.rst]: FYI, there's https://github.com/click-contrib/sphinx-click to update this based on the current output (I've never used it before).

Personally, I think manually updating this from time to time is fine.

File: docs/source/setup/hpc.rst [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+	lo          Link encap:Local Loopback                       # Localhost
+				inet addr:127.0.0.1  Mask:255.0.0.0
+				inet6 addr: ::1/128 Scope:Host
+	eth0        Link encap:Ethernet  HWaddr XX:XX:XX:XX:XX:XX   # Ethernet
+				inet addr:192.168.0.101
+				...
+	ib0         Link encap:Infiniband                           # Fast InfiniBand
+				inet addr:172.42.0.101
+
+https://stackoverflow.com/questions/43881157/how-do-i-use-an-infiniband-network-with-dask
```
Comment (TomAugspurger) [inline docs/source/setup/hpc.rst]: Does sphinx auto-covert this to a link?

File: docs/source/setup/kubernetes.rst [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+Although you don't need to use this address, the Dask client will find this
+variable automatically.
+
+.. code-block:: python
+   from dask.distributed import Client, config
+   client = Client()
+
+
+Configure Environment
```
Comment (TomAugspurger) [inline docs/source/setup/kubernetes.rst]: On thing I don't understand (as a newcomer to kubernetes) is what should the relationship be between nodes and workers? Say I have a cluster with 3 machines, what would a good number of workers be? Is it possible to answer that in general? Are there heuristics I could use to guess?

File: docs/source/setup.rst [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+   df = dd.read_csv(...)
+   df.x.sum().compute()  # This uses the single-machine scheduler by default
+
+.. code-block:: python
+
+   from dask.distributed import Client
+   client = Client(...)  # Connect to distributed cluster and override default
+   df.x.sum().compute()  # This now runs on the distributed system
+
+Note that the newer ``dask.distributed`` scheduler is often preferble even on
```
[the remark this answers is not in the document]
Comment (TomAugspurger) [reply docs/source/setup.rst]: I see that you mention this down in the dedicated distributed scheduler page, so perhaps it's not necessary to mention here.

File: docs/source/setup/kubernetes.rst (hunk 3 above)
Comment (mrocklin) [reply docs/source/setup/kubernetes.rst]: In general you don't think about this.  Kubernetes hides the size of your nodes from you.

File: docs/source/setup/hpc.rst (hunk 2 above)
Comment (mrocklin) [reply docs/source/setup/hpc.rst]: Yes

File: docs/source/setup/cli.rst (hunk 1 above)
Comment (mrocklin) [reply docs/source/setup/cli.rst]: Sticking with manual for now

File: docs/source/setup/cli.rst [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+
+Diagnostic Web Servers
+----------------------
+
+Additionally Dask schedulers and workers host interactive diagnostic web
+servers using `Bokeh <https://bokeh.pydata.org>`_.  These are optional, but
+generally useful to users.  The diagnostic server on the scheduler is
+particularly valuable, and is served on port ``8787`` by default (configurable
+with the ``-bokeh-port`` keyword).
```
[the remark this answers is not in the document]
Comment (mrocklin) [reply docs/source/setup/cli.rst]: Thanks, fixed!

File: docs/source/setup/cli.rst [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+
+
+Handling Ports
+--------------
+
+The scheduler and workers both need to accept TCP connections on an open port.
+By default the scheduler binds to port ``8786`` and the worker binds to a
+random open port.  If you are behind a firewall then you may have to open
+particular ports or tell Dask to listen on particular ports with the ``--port``
+and ``-worker-port`` keywords.::
```
[the remark this answers is not in the document]
Comment (mrocklin) [reply docs/source/setup/cli.rst]: Thanks, fixed!

**Pull request merged**